shared/fs/container.js

```diff
@@ -71,16 +71,21 @@ const ConnectedFiles = compose(
 )(Files)
 
 const FilesLoadingHoc = compose(
-  connect(), // provides `dispatch`
-  mapProps(({routeProps, dispatch}) => ({
-    path: routeProps.get('path', Constants.defaultPath),
+  connect(undefined, (dispatch: Dispatch) => ({
     loadFolderList: (path: Types.Path) => dispatch(FsGen.createFolderListLoad({path})),
   })),
+  mapProps(({routeProps, loadFolderList}) => ({
+    path: routeProps.get('path', Constants.defaultPath),
+    loadFolderList,
+  })),
   lifecycle({
     componentWillMount() {
       this.props.loadFolderList(this.props.path)
     },
-    componentWillUpdate(nextProps) {
+    componentWillReceiveProps(nextProps) {
+      if (this.props.path === nextProps.path) {
+        return
```

Comment:
You likely want the reverse of this. Only call if the oath is changed. If any other props change you don't want to call right

Reply:
Hmm this is to avoid loadFolderList when path *didn't* change. WDYM only call if the oath is changed?